ansible/configs/sap-smart/default_vars.yml

```diff
@@ -0,0 +1,531 @@
+project_tag: "{{ env_type }}-{{ guid }}"
+
+# This is where the ssh_config file will be created, this file is used to
+# define the communication method to all the hosts in the deployment
+deploy_local_ssh_config_location: "{{output_dir}}/"
+
+install_bastion: true
+install_common: true
+
+# Variable used to identify the request being made by a Red Hatter or Partner (external)
+rh_internal: true
+
+### If you want a Key Pair name created and injected into the hosts,
+# set `set_env_authorized_key` to true and set the keyname in `env_authorized_key`
+# you can use the key used to create the environment or use your own self generated key
+# if you set "use_own_key" to false your PRIVATE key will be copied to the bastion. (This is {{key_name}})
+use_own_key: true
+env_authorized_key: "{{guid}}key"
+#ansible_ssh_private_key_file: ~/.ssh/{{key_name}}.pem
+set_env_authorized_key: true
+
+# Is this running from Red Hat Ansible Tower
+tower_run: false
+
+## Networking
+subdomain_base_short: "{{ guid }}"
+subdomain_base_suffix: ".example.opentlc.com"
+subdomain_base: "{{subdomain_base_short}}{{subdomain_base_suffix}}"
+
+## Environment Sizing and OS
+
+bastion_instance_image: rhel-8.1
+bastion_instance_type: "sap-2-16"
+bastion_instance_count: 1
+rootfs_size_bastion: "{{ rootfs_size_bastion }}"
+
+hana_instance_image: rhel-8.1
+hana_instance_type: "sap-4-64"
+hana_instance_count: 1
+rootfs_size_hana: "{{ rootfs_size_hana }}"
+pv_size_hana: 500
+
+s4hana_instance_image: rhel-8.1
+s4hana_instance_type: "sap-4-32"
+s4hana_instance_count: 1
+rootfs_size_s4hana: "{{ rootfs_size_s4hana }}"
+pv_size_s4hana: 200
+
+tower_instance_image: rhel-8.1
+tower_instance_type: "sap-4-32"
+tower_instance_count: 1
+rootfs_size_tower: "{{ rootfs_size_tower }}"
+
+sat_instance_image: rhel-server-7.7
+sat_instance_type: "sap-4-32"
+sat_instance_count: 1
+rootfs_size_sat: "{{ rootfs_size_sat }}"
+
+# SAP Software image and variables
+sap_software_image: sap-software-v1.2
+sap_software_size: "{{ sap_software_size }}"
+
+#Ansible Tower Variables
+ansible_tower_download_url: "https://releases.ansible.com/ansible-tower/setup/ansible-tower-setup-3.6.3-1.tar.gz"
+ansible_tower_epel_download_url: "https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm"
+
+# OSP Security Groups
+security_groups:
+  - name: HanaSG
+    rules:
+      - name: HanaSSHPrivate
+        description: "SSH private"
+        from_port: 22
+        to_port: 22
+        protocol: tcp
+        cidr: "0.0.0.0/0"
+        rule_type: Ingress
+        group: BastionSG
+      - name: HanaUDPPortsPrivate
+        description: "Only from bastion"
+        from_port: 1
+        to_port: 65535
+        protocol: udp
+        group: BastionSG
+        rule_type: Ingress
+      - name: HanaTCPPPortsPrivate
+        description: "Only from bastion"
+        from_port: 1
+        to_port: 65535
+        protocol: tcp
+        group: BastionSG
+        rule_type: Ingress
+  - name: BastionSG
```

Comment:
Hmm, aren't these rules going to restrict incoming connections to the bastion even except from instances in the HanaSG group? Meaning you can't SSH to the bastion from the outside.

Reply:
No, external traffic to the bastion is open

Reply:
that doesn't make sense, but I think I see the problem. You are using an out of date version of the OSP cloud provider template. So you are getting what you have defined here PLUS what is defined in the defaults. If that is the case, I was not expecting that to be munged together when the same SG name is used. Hmmm. I guess if it works, you can keep it, but I'd recommend specifying the SSH access in your config for that SG to avoid unintended removal of it in the future.

https://github.com/redhat-cop/agnosticd/blob/development/ansible/roles-infra/infra-osp-template-generate/templates/cloud_template_master.j2#L58-L61